### PR TITLE
Use the etesync test-server image for CI

### PR DESCRIPTION
**Pull request by djmitche**
See https://github.com/etesync/server/issues/94.

If that is merged and the image pushed to `etesync/test-server`, then I'll update the workflow here to use it, and this will be mergeable.

### CODE REVIEW

**Comment by gchaincl** [comment]
LGTM, @tasn ?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,13 +20,13 @@ jobs:
...
-          docker run -d -v $PWD/scripts:/etebase/scripts -e DEBUG=true -p 80:3735 victorrds/etebase ./scripts/testmode.sh
-          ./scripts/wait-for-it.sh localhost:80
+          #docker run -p 80:3735 etesync/test-server:0.7.0
+          docker run -d -p 3735:3735 djmitche/etesync-test-server:0.7.0
+          ./scripts/wait-for-it.sh localhost:3735
           
       - name: Tests
         env:
-          ETEBASE_TEST_HOST: localhost:80
+          ETEBASE_TEST_SERVER: localhost:3735
```
Comment (tasn) [inline .github/workflows/build.yml:29]: Why rename the env var?

**Review by tasn — CHANGES_REQUESTED** [review]
Made one comment, but looks good to me otherwise!

File: .github/workflows/build.yml (hunk 1 above)
Comment (djmitche) [reply .github/workflows/build.yml:29]: I spent a bit trying to figure out how to customize the port, before realizing that this was not just a hostname.  I can change it back..

**Comment by djmitche** [comment]
Hm, this appears to fail against latest:
```
personal-devel ~/p/etebase-go [etesync-issue-94] $ ETEBASE_TEST_HOST=localhost:3735 go test .
--- FAIL: TestEtebaseSuite (0.05s)
    client.go:100: POST   /api/v1/authentication/signup/
    etebase_test.go:47: 
                Error Trace:    etebase_test.go:47
                                                        suite.go:118
                                                        etebase_test.go:116
                Error:          Received unexpected error:
                                msgpack: unexpected code=49 decoding map length
                Test:           TestEtebaseSuite
FAIL
FAIL    github.com/etesync/etebase-go   0.051s
FAIL
```

**Comment by djmitche** [comment]
(I see the same failure in CI)

**Comment by tasn** [comment]
What's the error onthe server side? What's the actual content returned from the server?

**Comment by djmitche** [comment]
It looks like this was due to a missing /reset endpoint due to the server not being in debug mode.  I assume it returned HTML in that case?  Anyway, this works again with the latest image.

**Comment by tasn** [comment]
Yeah, probably returned a 404 and html.

**Comment by djmitche** [comment]
(force-pushed with the real docker image name)

**Comment by tasn** [comment]
Seem to hang? Maybe need to make the docker containers run in the background?

**Comment by djmitche** [comment]
Whoops, yes.

**Pull request merged**